ocm_python_wrapper/cluster.py

```diff
@@ -256,33 +257,100 @@ def addon_info(self):
             self.addon_name
         ).to_dict()
 
-    def validate_addon_parameters(self, parameters):
-        _info = self.addon_info()
-        _parameters = _info.get("parameters")
-        if not _parameters and parameters:
-            raise ValueError(f"{self.addon_name} does not take any parameters")
+    def filter_required_addon_parameters(self, addon_parameters):
+        """Filters out addon parameter by specified cluster-related conditions.
 
+        Args:
+            addon_parameters (dict) : Addons parameters from api
+
+        Returns: List of dicts of parameters that filtered by matching clusters current configuration and are
+        required for installation.
+        """
+        filtered_parameters = []
         required_parameters = [
-            param["id"] for param in _parameters["items"] if param["required"] is True
+            {
+                "id": param["id"],
+                "default": param["default_value"]
+                if "default_value" in param.keys()
+                else None,
+                "conditions": param["conditions"]
+                if "conditions" in param.keys()
+                else None,
+            }
+            for param in addon_parameters["items"]
+            if ("required" in param.keys() and param["required"])
         ]
-        user_addon_parameters = [param["id"] for param in parameters]
 
+        for parameter in required_parameters:
+            for param_condition in parameter["conditions"]:
+                if param_condition["resource"] == "cluster":
+                    validation_flag = True
+                    for condition, condition_value in param_condition["data"].items():
+                        cluster_value = benedict(
+                            self.instance.to_dict(), keypath_separator="."
+                        ).get(condition)
+                        # Using logical nor to check if cluster value doesn't meet all the condition requirements.
+                        if not (
+                            (
+                                isinstance(condition_value, list)
+                                and (cluster_value in condition_value)
+                            )
+                            or cluster_value == condition_value
+                        ):
+                            validation_flag = False
+                            break
+                    if validation_flag:
```

Comment:
I s`if validation_flag:` needed? you break when `validation_flag` is false

Reply:
The reason is I have to know if the for loop finished or not.
I will break if There's a condition that my cluster have not met, so I want to add this parameter to the list only after passing all param x conditions -> complete passing the for loop. 

So I need an indication if the for loop had broken -> if validation flag (is true) - passed the loop.

Reply:
Please use `for-else` instead.
See https://www.w3schools.com/python/gloss_python_for_else.asp
```
for ...:
   for ....:
       if ...:
           break
    else:
       ....
     break
```

Reply:
Done